### PR TITLE
refactor: pipeline service to pb provider

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
refactor pipeline service to pb provider

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   efactor pipeline service to pb provider           |
| 🇨🇳 中文    |  将pipeline大的service重构成pb provider            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: cmd/pipeline/bootstrap.yaml [hunk 1]
```diff
@@ -49,7 +49,6 @@ edgepipeline_register:
   cluster_access_key: "${CLUSTER_ACCESS_KEY}"
   allowed_sources: "cdp-,fdp-"
   cluster_access_key_etcd_prefix: "${EDGE_PIPELINE_CLUSTER_ACCESS_KEY_ETCD_PREFIX:/devops/pipeline/v2/edge/cluster-access-key}"
-edgepipeline: {}
```
Comment (sfwn) [inline cmd/pipeline/bootstrap.yaml:52]: why remove edgepipeline provider?

File: internal/apps/cmp/impl/clusters/add_edge_cluster.go [hunk 2]
```diff
@@ -244,19 +249,20 @@ func (c *Clusters) MonitorCloudCluster() (abort bool, err error) {
...
-		for _, stage := range dto.PipelineStages {
+		for _, stage := range dto.Data.PipelineStages {
 			if len(stage.PipelineTasks) == 0 {
 				err = fmt.Errorf("empty task in pipeline stage")
 				_ = c.processFailedPipeline(ctx, record, err)
 				break
 			}
 			for _, task := range stage.PipelineTasks {
-				if task.Status.IsFailedStatus() {
+				taskStatus := apistructs.PipelineStatus(task.Status)
```
Comment (sfwn) [inline internal/apps/cmp/impl/clusters/add_edge_cluster.go:264]: move pipeline status into one of pipeline provider.

File: pkg/metadata/metadata.go [hunk 3]
```diff
@@ -90,3 +92,14 @@ func (metadata Metadata) FilterNoErrorLevel() (notErrorLevel, errorLevel Metadat
 	}
 	return
 }
+
+func FilterNoErrorLevel2PB(metas []*commonpb.MetadataField) (notErrorLevel, errorLevel []*commonpb.MetadataField) {
```
Comment (sfwn) [inline pkg/metadata/metadata.go:96]: the input is already pb, so the function name should be `FilterNoErrorLevelMeta`

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/pipeline/bootstrap.yaml (hunk 1 above)
Comment (chengjoey) [reply cmd/pipeline/bootstrap.yaml:52]: edgepipeline provider should be removed to resolve circular dependencies

File: pkg/metadata/metadata.go (hunk 3 above)
Comment (chengjoey) [reply pkg/metadata/metadata.go]: adjusted

File: internal/apps/cmp/impl/clusters/add_edge_cluster.go (hunk 2 above)
Comment (chengjoey) [reply internal/apps/cmp/impl/clusters/add_edge_cluster.go:264]: apistructs of pipeline will be moved in next pr

**Review by sfwn — APPROVED** [review]

**Pull request merged**